### ISSUE TITLE
Doubt in Tissue Fraction Quantification

### ISSUE THREAD

**Issue by Rukhmini** [Biooptics2021/PathFinder#8]
Hello,
Thank you for all the responses. I have another doubt in the tissue fraction quantification in Eq.23 of the paper. Did you use the segmentation map or the probability map (NEC heatmaps) for the necrosis channel to calculate Eq.23? I think for Eq.25 you have used the probability heatmaps (TND heatmaps).

**Comment by LiangJunhao-THU** [comment]
Hi @Rukhmini ,
Thank you for your question. Both TND and NEC are calculated based on segmentation map. Because we need first get the segmentation map **S** and know the category of each pixel, then we can perform calculation. 



In TIL or TND, one thing may cause confusion is the _p_ in the Eq. 21,22,24,25. They are the percentage of one category (e.g. tumor) in a grid-cell based on segmentation map **S**, not the raw probability predicted by PaSegNet.
<img width="420" alt="截屏2023-12-02 11 20 20" src="https://github.com/Biooptics2021/PathFinder/assets/41545346/963979fc-8009-4eec-81ec-963703e61e53">



The TND calculation code is also based on segmentation map.
`multi_classes[i][j] = np.argmax(pred_map[i,j,:])`
<img width="676" alt="截屏2023-12-02 11 22 09" src="https://github.com/Biooptics2021/PathFinder/assets/41545346/a95581ae-c9e4-4a9f-8f57-eabd79ae0ae6">

**Comment by Rukhmini** [comment]
Thanks for the response. I understand your explanation, I got confused because in the Co_loc function the input is mentioned as prob_map and also in TND_score.py file the input of the TND_Score function is (prob_map_path) where it is mentioned that "prob_map:MxNx8 numpy array contains the probabilities"
![image](https://github.com/Biooptics2021/PathFinder/assets/49904824/06e45804-070c-45ef-b1a4-be678137f774)
![image](https://github.com/Biooptics2021/PathFinder/assets/49904824/51209684-7a94-42ff-9aab-1743ecc35f5d)
Just to make sure, that means you have calculated all these features from the segmentation maps not from the heatmaps, right? Because, In Fig.9 and Fig.10 of your paper you showed both segmentation maps and the heatmaps for TND and NEC respectively. I got confused from those figures as well. That means, all these scores are calculated from the segmentation maps shown in Fig.3(a) for example. Also, another doubt is that, how did you choose the threshold for these TND and NEC features to do the KM-plot as shown in verification.ipynb file?

**Comment by LiangJunhao-THU** [comment]
Sorry for the confusion. In fact the segmentation map is also calculated based on the prob_map, that's why the input of the code is (prob_map_path).  As for extended figure 9 and 10, both segmentation maps and the heatmaps for TND and NEC are aimed to show different CNNs can get slightly different segmentation results, which lead to different TND heatmap. 
The best threshold for these TND and NEC in KM-plot is calculated by looping through threshold values. You can also select the medium value as the threshold, which is also significant.

**Comment by Rukhmini** [comment]
Thank you. Another doubt is that, did you use Co_loc function or TND_Score function to calculate the TND score?

**Comment by Rukhmini** [comment]
Hello,
How did you do the hazard ratio plot in Supplementary Fig.5. Did you train the model again with NEC and TND feature and then perform the univariable and multivariable analysis?

**Comment by LiangJunhao-THU** [comment]
Hi @Rukhmini ,
Here we showed a demo about using the newly founded biomarkers (TND, NEC) to build MultiModal transparent prognostic model (only with COX regression). So no deep learning model was trained again, only a COX prognosis model was generated with TND or NEC and other already known biomarkers.

**Comment by Rukhmini** [comment]
Thanks for the response Junhao. That means you calculated NEC and TND using Eq.23 and 25 and then use that calculated NEC and TND scores as features to stratify the patients into high risk and low risk.

**Comment by LiangJunhao-THU** [comment]
Yes. At this step NEC and TND is already identified and verified as the new significant biomarkers, just like AFP or TIL. Not like risk scores predicted by deep learning model (black box), they are white indicators which has clear pathological meaning and mathematical models. We show their potential in MultiModal transparent prognosis.

**Comment by Rukhmini** [comment]
How did you choose the threshold values? You mentioned that by looping through the threshold values you choose the threshold, from where did you get the threshold values to stratify the patients? Are these threshold obtained from cox prognosis model as cut-off values?

**Comment by LiangJunhao-THU** [comment]
The threshold for each variance (e.g. TND, NEC, TIL) used in multi variance analysis is the same as single variance analysis, which is selected based on the most significant p value in the single variance analysis. You can also select the median value, which is also significant and easy to choose.

**Comment by Rukhmini** [comment]
Thanks for the clarification.
